pkg/report/sarif_test.go

```diff
@@ -541,6 +541,176 @@ func TestReportWriter_Sarif(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "report with vulnerabilities and package locations",
+			input: types.Report{
+				ArtifactName: "debian:9",
+				ArtifactType: ftypes.ArtifactContainerImage,
+				Metadata: types.Metadata{
+					RepoTags: []string{
+						"debian:9",
+					},
+					RepoDigests: []string{
+						"debian@sha256:a8cc1744bbdd5266678e3e8b3e6387e45c053218438897e86876f2eb104e5534",
+					},
+				},
+				Results: types.Results{
+					{
+						Target: "library/test",
+						Class:  types.ClassOSPkg,
```

Comment:
IIRC OS package don't have locations

Reply:
Not sure, I copied the first test case and used it, it has location fields in packages but if you want I change it as well. Also it will be great if you can point out any relevant reading materials that I can go through.

Reply:
You can see the packet parser [here](https://github.com/aquasecurity/trivy/tree/main/pkg/fanal/analyzer/pkg).

You can see the language files that support places [here](https://aquasecurity.github.io/trivy/v0.50/docs/coverage/language/)

Reply:
> I copied the first test case and used it,

I think it's better to add language package (like yarn.lock or pom.xml) to avoid confusion when we support this code later.

Reply:
I have added a YARN vulnerability as a test case replacing the previous OSDependency